core/Cookie.php

```diff
@@ -269,25 +284,18 @@ protected function loadContentFromCookie($validateSignature = true)
      */
     public function generateContentString()
     {
-        $cookieStr = '';
+        $cookieStrArr = [];
 
         foreach ($this->value as $name => $value) {
-            if (!is_numeric($value)) {
-                $value = base64_encode(safe_serialize($value));
+            if (!is_numeric($value) && !is_string($value))  {
+                throw new \Exception('Only strings and numbers can be used in cookies. Value is of type ' . gettype($value));
+            } elseif (!is_numeric($value)) {
+                $value = base64_encode($value);
             }
-
-            $cookieStr .= "$name=$value" . self::VALUE_SEPARATOR;
-        }
-
-        if (!empty($cookieStr)) {
-            $cookieStr .= '_=';
-
-            // sign cookie
-            $signature = sha1($cookieStr . SettingsPiwik::getSalt());
-            return $cookieStr . $signature;
+            $cookieStrArr[] = "$name=$value"; 
```

Comment:
Is there a reason to not use json_encode? Don't think it really matters if this works, just wondering.

Reply:
probably no particular reason I guess.